configure

```diff
@@ -581,104 +1093,104 @@ return __builtin_popcountl(argc) == 100;
-      build/test-popcnt > /dev/null 2>&1
-      if [ $? -eq 0 ]; then
-         printf "%s\n" "yes"
-         echo "yes" >> config.log
-	 POPCNT_FLAG="-mpopcnt"
-      else
-         printf "%s\n" "no"
-	 echo "no" >> config.log
-      fi
-      rm -f build/test-popcnt{,.c}
-   #in case -mpopcnt is not available, the test program will use an illegal
-   #instruction and that will print out something on stderr when the if
-   #construction is exited, whence the following "2> /dev/null"
-   fi 2> /dev/null
+    printf "%s\n" "yes"
+    echo "yes" >> config.log
+    CONFIG_POPCNT_INTRINSICS="#define FLINT_USES_POPCNT"
+
+    if [ "$BUILD_MACHINE" = "x86_64" ]; then
+        MSG="Testing native popcount..."
+        printf "%s" "$MSG"
+        printf "%s" "$MSG" >> config.log
+        touch build/test-popcnt.c
+        rm build/test-popcnt
+        $CC -mpopcnt build/test-popcnt.c -o ./build/test-popcnt > /dev/null 2>&1
+        build/test-popcnt > /dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            printf "%s\n" "yes"
+            echo "yes" >> config.log
+            POPCNT_FLAG="-mpopcnt"
+        else
+            printf "%s\n" "no"
+            echo "no" >> config.log
+        fi
+        rm -f build/test-popcnt{,.c}
+        #in case -mpopcnt is not available, the test program will use an illegal
+        #instruction and that will print out something on stderr when the if
+        #construction is exited, whence the following "2> /dev/null"
+    fi 2> /dev/null
 else
-   rm -f build/test-popcnt.c
-   printf "%s\n" "no"
-   echo "no" >> config.log
+    rm -f build/test-popcnt.c
+    printf "%s\n" "no"
+    echo "no" >> config.log
 fi
 
 #defaults for CFLAGS
 if [ -z "$CFLAGS" ]; then
-   if [ "$OS" = "MINGW64" ]; then
-      CFLAGS="-std=c99 -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif [ "$OS" = "CYGWIN64" ]; then
-      CFLAGS="-O2 -funroll-loops -g -D _WIN64 $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif [ "$MACHINE" = "mips64" ]; then
-      CFLAGS="-O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-   elif test "$KERNEL" = "FreeBSD" -o "$OS" = "OpenBSD"; then
-      CFLAGS="-std=c99 -pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-      ANSI_FLAG=""
-      CXXFLAGS="-std=c++11 -pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-   else
-      ANSI_FLAG="-ansi"
-      CFLAGS="-pedantic -Wall -O2 -funroll-loops -g $POPCNT_FLAG $ABI_FLAG"
-   fi
+    if [ "$BUILD_OS" = "MINGW64" ]; then
+        CFLAGS="-std=c99 -O2 $POPCNT_FLAG $ABI_FLAG $DEBUGGING_FLAG $MARCH_FLAG"
+        ANSI_FLAG=""
+    elif [ "$BUILD_OS" = "CYGWIN64" ]; then
+        CFLAGS="-O2 -D _WIN64 $POPCNT_FLAG $ABI_FLAG $DEBUGGING_FLAG $MARCH_FLAG"
+        ANSI_FLAG=""
+    elif [ "$BUILD_MACHINE" = "mips64" ]; then
+        CFLAGS="-O2 $POPCNT_FLAG $ABI_FLAG $DEBUGGING_FLAG $MARCH_FLAG"
+        ANSI_FLAG=""
+    elif test "$BUILD_KERNEL" = "FreeBSD" -o "$BUILD_OS" = "OpenBSD"; then
```

Comment:
What's the difference between `BUILD_KERNEL` and `BUILD_OS`, and why is check using one for FreeBSD and the other for OpenBSD?!? Looking at the code which sets `BUILD_OS` above, could `BUILD_OS` be used for both, and also for the Darwin/macOS check below, and then `BUILD_KERNEL` can be removed?

Reply:
Don't know. I'm not really comfortable with other operating systems than Linux, so I tried to keep it as it was.

Reply:
I think both OS and KERNEL was there before, so I kept it.

Reply:
You mean it was there in Flint before, or are you taking this code from somewhere else? Please ensure the license is compatible if the latter.